google/cloud/storage/internal/curl_download_request.h

```diff
@@ -48,12 +48,14 @@ extern "C" std::size_t CurlDownloadRequestHeader(char* contents,
  */
 class CurlDownloadRequest : public ObjectReadSource {
  public:
-  explicit CurlDownloadRequest();
+  CurlDownloadRequest(CurlHeaders headers, CurlHandle handle, CurlMulti multi);
 
   ~CurlDownloadRequest() override;
 
-  CurlDownloadRequest(CurlDownloadRequest&&) = default;
-  CurlDownloadRequest& operator=(CurlDownloadRequest&& rhs) = default;
+  CurlDownloadRequest(CurlDownloadRequest&&) = delete;
+  CurlDownloadRequest& operator=(CurlDownloadRequest&&) = delete;
+  CurlDownloadRequest(CurlDownloadRequest const&) = delete;
```

Comment:
I suspect that this class used to be copyable. Do you think that may have been the issue? A copy showing up where it wasn't expected?

Reply:
Bummer. There's also the `buffer_` member, which seems like a `=default` move c'tor may not do the right thing with.

Regardless, I like the change of making this not moveable or copyable and avoiding all these guesses.